### ISSUE TITLE
Privileged containers cause host /home to unmount

### ISSUE THREAD

**Issue by elitegreg** [lxc/lxd#3426]
* Distribution: Ubuntu
 * Distribution version: 17.04
 * The output of "lxc info" or if that fails:
   * Kernel version: 4.10.0-22-generic
   * LXC version: 2.0.7
   * LXD version: 2.12
   * Storage backend in use: zfs

# Issue description

For quite a few months, I have had a container running in privileged mode sharing /home from the host. The other day after reboot, /home on the host was not mounted but it was mounted in the container. I stopped that container, but when I started another privileged container, the /home volume on the host was unmounted.

I'm using a zfs pool "storage-pool" as the backend for lxd. My home volume on the host is storage-pool/home.

**Comment by brauner** [comment]
Can you please show:

    lxc config show <container-name>
    cat /var/log/lxd/<container-name>/lxc.conf
    lxc profile show default

**Comment by stgraber** [comment]
Would be interesting to know if /home on the host is getting unmounted as a result of the container starting or as a result of the container stopping/rebooting.

I have a theory why the latter may happen (zfs bug) and that'd be limited to privileged containers, which explains why we didn't see widespread reports of this.

**Comment by elitegreg** [comment]
Hello,

I have found and fixed the issue. I will describe it briefly here in case someone else runs into it.

I'm exporting /home with NFSv4 which requires that I bind mount /home -> /exports/home. To do this, I had a bind mount entry in fstab. It turns out, this doesn't work correctly with ZFS on Linux. However, it does give the impression that it works. See, I was correctly seeing my home volume mounted at /exports/home (before starting a privileged container), so I was under the impression that the bind mount was working. But then I noticed this:

`# mount | grep home
/dev/sdb3 on /exports/home type ext4 (rw,relative,discard,data=ordered)
storage-pool/home on /home type zfs (rw,relatime,xattr,noacl)
storage-pool/home on /exports/home type zfs (rw,relatime,xattr,noacl)
`
The latter two entries are expected, however the former looks weird. It turns out, that the bind mount was being created before /home was mounted from zfs. Therefore, I was binding an empty /home from the root device to /exports/home. Somehow, after mounting ZFS, it figured out to bind mount itself to /exports/home and everything looked good, at least all the way up until the privileged container messed it up. Now I'm not sure exactly what happens with the container, however, the issue is 100% resolved by removing the bind mount from fstab and creating a systemd service file to do the bind mount after zfs before nfs.

Thanks!
